File: tests/vb/type_gettype_dynamic/get_type_resolves_a_user_declared_class.vb
' vybe-test: vb/type_gettype_dynamic/get_type_resolves_a_user_declared_class
' origin: languages/vb/tests/vb/test_type_gettype_dynamic.rs

' Vybe test harness — Visual Basic.
'
' Real VB source alongside harness/go/check.go and harness/js/check.js, the way
' test262's assert.js is JavaScript.
'
' A test's verdict is its EXIT CODE. __Check prints its diagnostic BEFORE
' throwing: an uncaught exception surfaces as `RuntimeError: [object]`, which
' says nothing at all.

Module VybeCheck
    Sub __Check(got As String, want As String)
        If got <> want Then
            Console.WriteLine("FAIL: want [" & want & "] got [" & got & "]")
            Throw New Exception("assertion failed")
        End If
    End Sub
End Module

Class Widget
End Class

Module Program
    Sub Main()
        __Check(CStr(Type.GetType("Widget") IsNot Nothing), "True")
    End Sub
End Module

File: tests/csharp/type_gettype_dynamic/get_type_resolves_a_user_declared_class.cs
// vybe-test: csharp/type_gettype_dynamic/get_type_resolves_a_user_declared_class
// origin: languages/csharp/tests/csharp/test_type_gettype_dynamic.rs

using static __Harness;

__P("Valid_get_type_resolves_a_user_declared_class");
__Check("Valid_get_type_resolves_a_user_declared_class");
public static class __Harness {
    public static string __buf = "";
    public static void __P(string s) { __buf = __buf + s + "\n"; }
    public static void __Pr(string s) { __buf = __buf + s; }
    public static void __Check(string want) {
        if (__buf != want && __buf != want + "\n") {
            Console.WriteLine("FAIL: want [" + want + "] got [" + __buf + "]");
            throw new Exception("assertion failed");
        }
    }
}
